src/hooks/useBestTrade.test.ts

```diff
@@ -140,7 +149,12 @@ describe('#useBestV3Trade ExactOut', () => {
 
     const { result } = renderHook(() => useBestTrade(TradeType.EXACT_OUTPUT, DAIAmount, USDC_MAINNET))
 
-    expect(mockUseRoutingAPITrade).toHaveBeenCalledWith(TradeType.EXACT_OUTPUT, undefined, USDC_MAINNET)
+    expect(mockUseRoutingAPITrade).toHaveBeenCalledWith(
```

Comment:
@Zach Pomerantz i would recommend switching useRoutingApiTrade to take a params object instead of ordered arguments, then allow defaults again (for example, defaulting to CLIENT)

Reply:
That feels like a bigger change, and switching to a params object risks perf issues with hooks memoization.
The RouterPreference is always available, so I'd rather just make that a new required argument and take it out of useRoutingApiTrade.